### PR TITLE
Fix small bugs

### PR DESCRIPTION
**Pull request by sandlerben**
These were all fixes to problems that occurred in production with IdleFreePhilly.

### CODE REVIEW

**Comment by sandlerben** [comment]
r @abhisuri97

File: app/templates/account/reset_password.html [hunk 1]
```diff
@@ -6,6 +6,7 @@
         <div class="eight wide computer sixteen wide mobile centered column">
             <h2 class="ui dividing header">Reset your password</h2>
             {{ f.render_form(form) }}
+            {{ form.hidden_tag() }}
```
Comment (aharelick) [inline app/templates/account/reset_password.html]: doesn't this get called by the `begin_form` macro that happens when you say `f.render_form(form)`?

File: app/__init__.py [hunk 2]
```diff
@@ -19,7 +19,9 @@
 compress = Compress()
 # Set up Flask-Login
 login_manager = LoginManager()
-login_manager.session_protection = 'strong'
+# TODO: Ideally this should be strong, but that led to bugs. Once this is
+# fixed, switch protection mode back to 'strong'
+login_manager.session_protection = 'basic'
```
Comment (aharelick) [inline app/__init__.py:24]: what bugs were occurring and do you know why this fixed them?

Comment (abhisuri97) [inline app/__init__.py:24]: Session protection works by generating a key-value pair of a random hashed string on the user end and in the SESSION variable. If these are not equal, in `basic` mode, only routes which require a `fresh` session (i.e. not a remember_me session for example) will ask the user to reauthenticate. But we don't implement a check from `fresh` login on our routes. In `strong` mode, the user is forced to reauthenticate when the SESSION and user hashed string do not match _no matter what_. Hence it is best to remain at `strong` session_protection level.

File: app/templates/account/reset_password.html (hunk 1 above)
Comment (abhisuri97) [inline app/templates/account/reset_password.html]: That is correct. Note you are essentially including two copies of a CSRF token.

**Comment by abhisuri97** [comment]
set session protection back to strong and remove the redundant hidden_tag. Merge when those are completed (unless you have objections).

File: app/templates/account/reset_password.html (hunk 1 above)
Comment (sandlerben) [inline app/templates/account/reset_password.html]: Oh... weird. This definitely fixed a bug in cac but I'll remove it from this PR.

**Comment by sandlerben** [comment]
Strong session protection led to problems in production (users were logged off after minutes). Anyone is welcome to look into that problem but in the meantime let's remove the bug from flask-base.

**Comment by abhisuri97** [comment]
I have not seen this bug on RTM. But I don't necessarily see any objections to switching to `basic` session protection. So, lgtm (as soon as you remove `{{ form.hidden_tag() }}`).

**Comment by abhisuri97** [comment]
did the commit. merged.

**Pull request merged**